### PR TITLE
opam-libs: update checksum

### LINKED ISSUE #12854: Wrong checksums for opam
Hi,

The checksums (hashes, really) for opam don't match those in the archive:

```
[ERROR] The sources of the following couldn't be obtained, aborting:
          - opam-client.2.0.1: Bad checksum
          - opam-core.2.0.1: Bad checksum
          - opam-format.2.0.1: Bad checksum
          - opam-repository.2.0.1: Bad checksum
          - opam-solver.2.0.1: Bad checksum
          - opam-state.2.0.1: Bad checksum
```

Thanks!

### PR DESCRIPTION
**Pull request by rjbou**
closes #12854

### CODE REVIEW

File: packages/opam-devel/opam-devel.2.0.1/opam [hunk 1]
```diff
@@ -33,12 +33,12 @@ The development version of opam has been successfuly compiled into %{lib}%/%{nam
 If you just want to give it a try without altering your current installation, you could use instead:
     alias opam2="OPAMROOT=~/.opam2 %{lib}%/%{name}%/opam\""""
     {success}
-dev-repo: "git+https://github.com/ocaml/opam.git"
+dev-repo: "git+https://github.com/ocaml/opam.gitd71c362d70a6b6cb9fe54cc43bfbea8079cca05d4721a7dda951196f543a1ff5af81399a246e0cd68bc590c7bdca388d96c7127217ed637261957047ce725516"
```
Comment (mseri) [inline packages/opam-devel/opam-devel.2.0.1/opam]: Is this change correct?

**Comment by mseri** [comment]
Why were the checksum broken?

File: packages/opam-devel/opam-devel.2.0.1/opam (hunk 1 above)
Comment (rjbou) [reply packages/opam-devel/opam-devel.2.0.1/opam]: wrong sed...

**Comment by rjbou** [comment]
After  #12826 push, we had to regenerate an archive to fix an alpine/patch problem, but forgot to update checksums.

**Comment by mseri** [comment]
I see

**Pull request merged**